Redo extra groups and all and add core (#7536)
sunpy/sunpy: changelog/7536.feature.rst

Added a "core" extra group that does not install any truly optional dependencies.

> It only includes the dependencies that are required to import sunpy or any of its subpackages.

Reviewer: ```suggestion
It only includes the dependencies that are required to import sunpy and all subpackages.
```

This means it will not install:

- asdf
- glymur
- opencv
- scikit-image

> - spicepy

Reviewer: ```suggestion
- spiceypy
```
